doc/ipython-notebooks/gaussian_process/variational_classifier.ipynb

```diff
@@ -30,8 +37,51 @@
+     ]
+    },
+    {
+     "cell_type": "heading",
+     "level": 2,
+     "metadata": {},
+     "source": [
+      "<a id='toc'>Table of Content</a>"
+     ]
+    },
+    {
+     "cell_type": "markdown",
+     "metadata": {},
+     "source": [
+      "* <a href=\"#intro\">Brief Introduction</a>\n",
+      "* Theory (Skip if you just want code examples)\n",
+      " * <a href=\"#bigpic\">Big Picture of Variational Inference for GPC Models</a>\n",
+      " * <a href=\"#detailbg\">More Detailed Background about GPC Models</a>\n",
+      " * <a href=\"#VGI\">Variational Gaussian Inference for GPC Models</a>\n",
+      " * <a href=\"#nonclosedForm\">Dealing with the Non-closed Form Issue in GPC Models</a>\n",
+      " * <a href=\"#bgvb\">Some Background of Variational Bounds in GPC Models</a>\n",
+      " * <a href=\"#ref\">Reference</a>\n",
+      "* Summary (Shogun's C++ API) \n",
+      " * <a href='#likelihood'>Likelihood Classes for GPC Models in Shogun</a>\n",
+      " * <a href=\"#inference\">Inference Methods for GPC Models in Shogun</a>\n",
+      " * <a href='#opt'>Optimizers Used in Inference Methods (for advanced users)</a>\n",
+      " * <a href='#linesearch'>Line Search Methods Used in LBFGS Optimizer (for advanced users)</a>\n",
+      "* Code Examples\n",
+      " * <a href='#toy'>A Toy Example of Variational Gaussian Inference</a>\n",
+      " * <a href='#visu'>An Example for Visualization</a>\n",
+      "     * <a href=\"#largescale\">Inference Methods for sparse GPC Models (Large Scale Inference)</a> \n",
+      " * <a href=\"#realeg1\">A Real-World Example (sonar dataset)</a>\n",
+      " * <a href=\"#realeg2\">Another Real-world Example (USPS dataset)</a>\n",
+      "     * <a href='#laplace'>Using Laplace Method</a>\n",
+      "     * <a href='#covv'>Using Covariance Variational Method</a>\n",
+      "     * <a href='#meanfield'>Using Mean-field Variational Method</a>\n",
+      "     * <a href=\"#dual\">Using Dual Variational Method</a>\n",
+      "     * <a href=\"#chol\">Using Cholesky Variational Method</a>\n",
+      "     * <a href=\"#bounds\">Likelihood Classes Supported Variational Bounds</a> \n",
+      "     * <a href='#parameters'>Optimizing Parameters of GPC (Model Selection)</a>\n",
+      "* <a href='#todo'>Soon to Come</a>\n",
+      " * Scalable inference methods for GPC models (stochastic variational inference for sparse GPC models)\n",
+      " * Latent GP models for dimension reduction"
```

Comment:
I really really like this TOC. And it nicely showcases the massive amount of work that you have put into this